sslr-examples/src/main/java/org/sonar/sslr/examples/grammars/typed/JsonNodeBuilder.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SonarSource Language Recognizer
+ * Copyright (C) 2010 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.sslr.examples.grammars.typed;
+
+import com.sonar.sslr.api.Rule;
+import com.sonar.sslr.api.TokenType;
+import com.sonar.sslr.api.Trivia;
+import com.sonar.sslr.api.typed.Input;
+import com.sonar.sslr.api.typed.NodeBuilder;
+import java.util.List;
+import org.sonar.sslr.examples.grammars.typed.impl.InternalSyntaxToken;
+import org.sonar.sslr.grammar.GrammarRuleKey;
+
+public class JsonNodeBuilder implements NodeBuilder {
+
+  @Override
+  public Object createNonTerminal(GrammarRuleKey ruleKey, Rule rule, List<Object> children, int startIndex, int endIndex) {
+    for (Object child : children) {
+      if (child instanceof InternalSyntaxToken) {
+        return child;
+      }
+    }
+    return null;
+  }
```

Comment:
so this methods seems to be called for every rule defined in `JsonLexer` (i.e. `TRUE`, `NUMBER`, etc.)

these are supposed to be tokens in fact (as `createTerminal`), but are non-terminals due to whitespaces.
the whitespace token is dropped, and only the first one is returned